virl2_client/models/auth_management.py

```diff
@@ -217,14 +174,33 @@ def test_current_auth(
             "auth-config": current,
             "auth-data": {"username": username, "password": password},
         }
-        response = self._session.post(self._TEST_URL, json=body)
+        response = self._session.post(_TEST_URL, json=body)
         return response.json()
 
 
+class AuthMethodManager:
+    METHOD = ""
+
+    def __init__(self, auth_management: AuthManagement):
+        self._auth_management = auth_management
+
+    def _check_method(self):
+        """This makes sure that storing the manager in a variable won't allow the user
+        to accidentaly set settings of other methods using this."""
+        if self._auth_management._get_setting("method") != self.METHOD:
+            raise MethodNotActive(f"{self.METHOD} is not the currently active method.")
+
+    def _get_setting(self, setting):
```

Comment:
```suggestion
    def _get_setting(self, setting: str) -> Any:
```
this is possible too, but probably an overkill especially when considering that more types will be added in future 
```
    def _get_setting(self, setting: str) -> str | bool | ResourcePool
```

Reply:
Not following the discussion, but I don't think the auth method setting in the client should use the ResourcePool object itself, just its ID as a string. You never know if the object still exists.

Reply:
That, and also this wouldn't be really helpful. I guess it tells you that no settings ask for any other types, but you still don't know whether a setting needs string or bool or RP.

Reply:
resource_pool setter allows both RP and RP.id as value (look at the end of this file)